src/Components/Endpoints/src/Rendering/SSRRenderModeBoundary.cs

```diff
@@ -190,10 +190,19 @@ public ComponentMarker ToMarker(HttpContext httpContext, int sequence, object? k
         return marker;
     }
 
-    private string GenerateMarkerKey(int sequence, object? key)
+    private ComponentMarkerKey GenerateMarkerKey(int sequence, object? componentKey)
     {
         var componentTypeNameHash = _componentTypeNameHashCache.GetOrAdd(_componentType, ComputeComponentTypeNameHash);
-        return $"{componentTypeNameHash}:{sequence}:{(key as IFormattable)?.ToString(null, CultureInfo.InvariantCulture)}";
+        var sequenceString = sequence.ToString(CultureInfo.InvariantCulture);
+
+        var locationHash = $"{componentTypeNameHash}:{sequenceString}";
+        var formattedComponentKey = (componentKey as IFormattable)?.ToString(null, CultureInfo.InvariantCulture) ?? string.Empty;
```

Comment:
In your comment at https://github.com/dotnet/aspnetcore/pull/50848#issuecomment-1736054282, you mention that:

> An SSR update can only supply parameter updates to an interactive component if a non-null @key was specified

But the code here always coalesces null into `string.Empty`, so it will never be null, will it? Or do nulls some from some other code path?

Or is the rule actually based on the non-emptiness of the key as opposed to its nullness?

Reply:
Yes, it uses `string.IsNullOrEmpty` [here](https://github.com/dotnet/aspnetcore/blob/ca1388f73fda974e73bf3469f9e9423cf1fe7cce/src/Components/Shared/src/WebRootComponentManager.cs#L152) rather than just a `null` check.